NewEssentials/Players/AfkChecker.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Cysharp.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using NewEssentials.API.Players;
+using NewEssentials.Extensions;
+using OpenMod.API;
+using OpenMod.API.Eventing;
+using OpenMod.API.Ioc;
+using OpenMod.API.Plugins;
+using OpenMod.API.Users;
+using OpenMod.Core.Helpers;
+using OpenMod.Core.Users;
+using OpenMod.Core.Users.Events;
+using OpenMod.Unturned.Users;
+using SDG.Unturned;
+
+namespace NewEssentials.Players
+{
+    [ServiceImplementation]
+    public class AfkChecker : IAfkChecker, IAsyncDisposable
+    {
+        
+        //TODO: Add translations
+
+        private Dictionary<IUser, TimeSpan> m_Users;
+
+        private readonly IEventBus m_Events;
+        private readonly IConfiguration m_Config;
+        private readonly NewEssentials m_Plugin;
+        private readonly IUserManager m_UserManager;
+        private readonly IRuntime m_Runtime;
+        
+        public AfkChecker(IEventBus bus, IConfiguration config, IPluginAccessor<NewEssentials> plugin, IUserManager users, IRuntime runtime)
+        {
+            if (!config.GetValue<bool>("afkchecker:enabled"))
+                return;
+            m_Users = new Dictionary<IUser, TimeSpan>();
+            
+            bus.Subscribe<UserConnectedEvent>(runtime, (provider, sender, @event) => PlayerJoin(@event));
+            bus.Subscribe<UserConnectedEvent>(runtime, (provider, sender, @event) => PlayerLeave(@event));
+            
+            
+            AsyncHelper.Schedule("NewEssentials::AfkChecker", async () => await CheckAfkPlayers());
+            m_Events = bus;
+            m_Config = config;
+            m_Plugin = plugin.Instance;
+            m_UserManager = users;
+            m_Runtime = runtime;
+        }
+
+        private async Task PlayerJoin(UserConnectedEvent @event)
+        {
+            if (m_Users.ContainsKey((UnturnedUser)@event.User))
+            {
+                m_Users[(UnturnedUser)@event.User] = DateTime.Now.TimeOfDay;
+                return;
+            }
+            
+            m_Users.Add((UnturnedUser)@event.User, DateTime.Now.TimeOfDay);
+            PlayerMovementCheckerComponent component = ((UnturnedUser) @event.User).Player.gameObject.AddComponent<PlayerMovementCheckerComponent>();
+            component.Resolve(this);
+
+        }
+
+        private async UniTask CheckAfkPlayers()
+        {
+            while (m_Plugin.IsComponentAlive)
```

Comment:
use `IPluginAccessor<NewEssentials> m_PluginAccessor` and `m_PluginAccessor.Instance != null && m_PluginAccessor.IsComponentAlive` instead